projects/ad9656/common/ad9656_bd.tcl

```diff
@@ -0,0 +1,123 @@
+
+# RX parameters
+set RX_NUM_OF_LANES 4      ; # L
+set RX_NUM_OF_CONVERTERS 4 ; # M
+set RX_SAMPLES_PER_FRAME 1 ; # S
+set RX_SAMPLE_WIDTH 16     ; # N/NP
+
+set RX_SAMPLES_PER_CHANNEL 2 ; # L * 32 / (M * N)
+
+source $ad_hdl_dir/library/jesd204/scripts/jesd204.tcl
+
+# adc peripherals
+
+ad_ip_instance axi_adxcvr axi_ad9656_rx_xcvr
+ad_ip_parameter axi_ad9656_rx_xcvr CONFIG.NUM_OF_LANES $RX_NUM_OF_LANES
+ad_ip_parameter axi_ad9656_rx_xcvr CONFIG.QPLL_ENABLE 0
+ad_ip_parameter axi_ad9656_rx_xcvr CONFIG.TX_OR_RX_N 0
+ad_ip_parameter axi_ad9656_rx_xcvr CONFIG.SYS_CLK_SEL 0
+ad_ip_parameter axi_ad9656_rx_xcvr CONFIG.OUT_CLK_SEL 3
+
+adi_axi_jesd204_rx_create axi_ad9656_rx_jesd $RX_NUM_OF_LANES
+
+ad_ip_instance util_cpack2 util_ad9656_rx_cpack [list \
+  NUM_OF_CHANNELS $RX_NUM_OF_CONVERTERS \
+  SAMPLES_PER_CHANNEL $RX_SAMPLES_PER_CHANNEL \
+  SAMPLE_DATA_WIDTH $RX_SAMPLE_WIDTH \
+  ]
+
+adi_tpl_jesd204_rx_create rx_ad9656_tpl_core $RX_NUM_OF_LANES \
+                                               $RX_NUM_OF_CONVERTERS \
+                                               $RX_SAMPLES_PER_FRAME \
+                                               $RX_SAMPLE_WIDTH
+
+ad_ip_instance axi_dmac axi_ad9656_rx_dma
+ad_ip_parameter axi_ad9656_rx_dma CONFIG.DMA_TYPE_SRC 2
+ad_ip_parameter axi_ad9656_rx_dma CONFIG.DMA_TYPE_DEST 0
+ad_ip_parameter axi_ad9656_rx_dma CONFIG.CYCLIC 0
+ad_ip_parameter axi_ad9656_rx_dma CONFIG.SYNC_TRANSFER_START 1
+ad_ip_parameter axi_ad9656_rx_dma CONFIG.ASYNC_CLK_DEST_REQ 1
+ad_ip_parameter axi_ad9656_rx_dma CONFIG.ASYNC_CLK_SRC_DEST 1
+ad_ip_parameter axi_ad9656_rx_dma CONFIG.ASYNC_CLK_REQ_SRC 1
+ad_ip_parameter axi_ad9656_rx_dma CONFIG.DMA_2D_TRANSFER 0
+ad_ip_parameter axi_ad9656_rx_dma CONFIG.DMA_DATA_WIDTH_SRC [expr 32*$RX_NUM_OF_LANES]
+ad_ip_parameter axi_ad9656_rx_dma CONFIG.MAX_BYTES_PER_BURST 256
+ad_ip_parameter axi_ad9656_rx_dma CONFIG.AXI_SLICE_DEST true
+ad_ip_parameter axi_ad9656_rx_dma CONFIG.AXI_SLICE_SRC true
+
+# common cores
+
+
+ad_ip_instance util_adxcvr util_ad9656_xcvr
+ad_ip_parameter util_ad9656_xcvr CONFIG.RX_NUM_OF_LANES $RX_NUM_OF_LANES
+ad_ip_parameter util_ad9656_xcvr CONFIG.TX_NUM_OF_LANES 0
+ad_ip_parameter util_ad9656_xcvr CONFIG.TX_OUT_DIV 1
+ad_ip_parameter util_ad9656_xcvr CONFIG.CPLL_FBDIV 2
+ad_ip_parameter util_ad9656_xcvr CONFIG.CPLL_FBDIV_4_5 5
+ad_ip_parameter util_ad9656_xcvr CONFIG.RX_OUT_DIV 1
```

Comment:
With these parameters, what is the VCO frequency ? The valid range for GTH transceivers is 2GHz to 6.25GHz

Reply:
2.5 GHz